File: app/src/main/java/social/entourage/android/map/encounter/EncounterDisclaimerFragment.java
package social.entourage.android.map.encounter;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.DialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.widget.CheckBox;
import android.widget.Toast;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import social.entourage.android.R;

public class EncounterDisclaimerFragment extends DialogFragment {

    // ----------------------------------
    // Constants
    // ----------------------------------

    public static final String TAG = "social.entourage.android.entourage.prodisclaimer";

    // ----------------------------------
    // Attributes
    // ----------------------------------

    @BindView(R.id.encounter_disclaimer_checkbox)
    CheckBox acceptCheckbox;

    // ----------------------------------
    // Lifecycle
    // ----------------------------------

    private OnFragmentInteractionListener mListener;

    public EncounterDisclaimerFragment() {
        // Required empty public constructor
    }

    public static EncounterDisclaimerFragment newInstance() {
        return new EncounterDisclaimerFragment();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        if (getDialog() != null && getDialog().getWindow() != null) {
            getDialog().getWindow().requestFeature(Window.FEATURE_NO_TITLE);
        }
        super.onCreateView(inflater, container, savedInstanceState);
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_encounter_disclaimer, container, false);
        ButterKnife.bind(this, view);

        return view;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof OnFragmentInteractionListener) {
            mListener = (OnFragmentInteractionListener) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        if (getDialog() != null && getDialog().getWindow() != null && getDialog().getWindow().getAttributes() != null) {
            getDialog().getWindow().getAttributes().windowAnimations = R.style.CustomDialogFragmentSlide;
        }
    }

    // ----------------------------------
    // Button handling
    // ----------------------------------

    @OnClick(R.id.encounter_disclaimer_close_button)
    protected void onCloseClicked() {
        dismiss();
    }

    @OnClick(R.id.encounter_disclaimer_ok_button)
    protected void onOkClicked() {
        if (acceptCheckbox.isChecked()) {
            //inform the listener that the user accepted the CGU
            mListener.onEncounterDisclaimerAccepted(this);
        }
        else {
            Toast.makeText(getActivity(), R.string.encounter_disclaimer_error_notaccepted, Toast.LENGTH_SHORT).show();
        }
    }

    // ----------------------------------
    // Listener
    // ----------------------------------

    public interface OnFragmentInteractionListener {

        void onEncounterDisclaimerAccepted(EncounterDisclaimerFragment fragment);
    }
}

File: app/src/main/java/social/entourage/android/tour/encounter/EncounterDisclaimerFragment.kt
package social.entourage.android.tour.encounter

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.Window
import androidx.fragment.app.DialogFragment
import com.google.android.material.snackbar.Snackbar
import kotlinx.android.synthetic.main.fragment_encounter_disclaimer.*
import social.entourage.android.R
import social.entourage.android.tools.view.EntourageSnackbar

class EncounterDisclaimerFragment : DialogFragment() {
    // ----------------------------------
    // Attributes
    // ----------------------------------
    private var mListener: OnFragmentInteractionListener? = null

    // ----------------------------------
    // Lifecycle
    // ----------------------------------
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        dialog?.window?.requestFeature(Window.FEATURE_NO_TITLE)
        super.onCreateView(inflater, container, savedInstanceState)
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_encounter_disclaimer, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        encounter_disclaimer_close_button?.setOnClickListener {onCloseClicked()}
        encounter_disclaimer_ok_button?.setOnClickListener {onOkClicked()}
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context !is OnFragmentInteractionListener){
            throw RuntimeException(context.toString()
                    + " must implement OnFragmentInteractionListener")
        }
        mListener = context
    }

    override fun onDetach() {
        super.onDetach()
        mListener = null
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        dialog?.window?.attributes?.windowAnimations = R.style.CustomDialogFragmentSlide
    }

    // ----------------------------------
    // Button handling
    // ----------------------------------
    fun onCloseClicked() {
        dismiss()
    }

    private fun onOkClicked() {
        if (encounter_disclaimer_checkbox?.isChecked == true) {
            //inform the listener that the user accepted the CGU
            mListener?.onEncounterDisclaimerAccepted(this)
        } else {
            view?.let {EntourageSnackbar.make(it, R.string.encounter_disclaimer_error_notaccepted, Snackbar.LENGTH_SHORT).show()}
        }
    }

    // ----------------------------------
    // Listener
    // ----------------------------------
    interface OnFragmentInteractionListener {
        fun onEncounterDisclaimerAccepted(fragment: EncounterDisclaimerFragment)
    }

    companion object {
        // ----------------------------------
        // Constants
        // ----------------------------------
        const val TAG = "social.entourage.android.entourage.prodisclaimer"
    }
}